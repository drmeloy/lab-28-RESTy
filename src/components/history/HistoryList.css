.History {
  grid-area: history;
  border: 1px solid;
  margin: 10px;
  background: #ffffff4a;
  border-radius: 25px;

  h1 {
    text-align: center;
    font-family: 'Oswald', sans-serif;
  }
}