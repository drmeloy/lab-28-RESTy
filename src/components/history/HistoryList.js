import React from 'react';
import PropTypes from 'prop-types';
import HistoryItem from './HistoryItem';
import styles from './HistoryList.css';

const HistoryList = ({ historyList }) => {
  const historyItems = historyList.map(item => (    
    <li key={Math.random() * 1000}>
      <HistoryItem {...item} />
    </li>
  ));

  return (
    <div className={styles.History}>
      <h1>History</h1>
      <ul>
        {historyItems}
      </ul>
    </div>
  );
}

HistoryList.propTypes = {
  HistoryList: PropTypes.arrayOf(PropTypes.shape({
    method: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
    requestBody: PropTypes.string
  }))
}

export default HistoryList;
