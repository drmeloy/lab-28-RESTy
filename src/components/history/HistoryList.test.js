import React from 'react';
import { shallow } from 'enzyme';
import HistoryList from './HistoryList';

describe('HistoryList component', () => {
  it('renders HistoryList', () => {
    const wrapper = shallow(<HistoryList historyList={[{
      url: 'https://swapi.co/api/people/',
      method: 'GET',
      requestBody: ''
    }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
