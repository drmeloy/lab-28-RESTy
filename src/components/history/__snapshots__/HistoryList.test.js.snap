// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryList component renders HistoryList 1`] = `
<div
  className="History"
>
  <h1>
    History
  </h1>
  <ul>
    <li
      key="991.9282160463782"
    >
      <HistoryItem
        method="GET"
        requestBody=""
        url="https://swapi.co/api/people/"
      />
    </li>
  </ul>
</div>
`;
